tests/ui/deprecations.rs

```diff
@@ -72,3 +72,19 @@ fn module_bound_by_value(m: Bound<'_, PyModule>) -> PyResult<()> {
     m.add_function(wrap_pyfunction!(double, &m)?)?;
     Ok(())
 }
+
+fn test_wrap_pyfunction(py: Python<'_>, m: &Bound<'_, PyModule>) {
+    // should lint
+    let _ = wrap_pyfunction!(double, py);
+
+    // should lint but currently does not
+    let _ = wrap_pyfunction!(double)(py);
```

Comment:
Yeah I suspect because of type inference within the closure this one will be hard to make work. But maybe there's a smart way!

Reply:
Yeah, it kept inferring from the inside instead of the outside.